turn_alerts/tests/test_views.py

```diff
@@ -0,0 +1,384 @@
+                            "uuid": "4e29c576-5901-4aaa-22bb-9887e0c5bc98",
+                        }
+                    },
+                    "from": "2349039756628",
+                    "id": "wamid.HBgNMjM0OTAzOTc1NjYyOBUCABIYFDNGODg1NDQxRDE0MjFGOURBOUY1AA==",
+                    "text": {"body": "Testing"},
+                    "timestamp": "1735424273",
+                    "type": "text",
+                }
+            ],
+        }
+
+        response = self.client.post(
+            reverse("turn-messages", kwargs={"org_id": self.org.id}),
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+            HTTP_X_TURN_HOOK_SUBSCRIPTION="whatsapp",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class TurnAlertsLayerViewTest(APITestCase):
+
+    def setUp(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        self.org = create_org()
+        self.turnalerts_account = create_turnalerts(org=self.org)
+
+    def generate_hmac_signature(self, data, key):
+        data = JSONRenderer().render(data)
+        h = hmac.new(key.encode(), data, sha256)
+        return base64.b64encode(h.digest()).decode()
+
+    @patch("turn_alerts.views.start_turn_journey")
+    def test_start_turn_journey_called_with_correct_arguments(
+        self, mock_start_turn_journey
+    ):
+        request_data = {
+            "statuses": [
+                {
+                    "errors": [
+                        {
+                            "code": 131026,
+                            "error_data": {"details": "Message Undeliverable."},
+                            "message": "Message undeliverable",
+                            "title": "Message undeliverable",
+                        }
+                    ],
+                    "id": "wamid.HBgFMzE1MzEVAgARGBIzMzQ2RUY0MUU4OTJGOEM5MjAA",
+                    "recipient_id": "2779631245",
+                    "status": "failed",
+                    "timestamp": "1735425598",
+                }
+            ]
+        }
+        turn_action = Mock()
+        turn_action.journey_id = "journey_id"
+        turn_action.error_code = 131026
+
+        rf = APIRequestFactory()
+        request = rf.post(
+            "/1/api/v2/messages",
+            request_data,
+            format="json",
+        )
+
+        secret_key = "REPLACEME"
+        generated_signature = self.generate_hmac_signature(request_data, secret_key)
+        request.META["HTTP_X_TURN_HOOK_SIGNATURE"] = generated_signature
+        request.META["HTTP_X_TURN_HOOK_SUBSCRIPTION"] = "whatsapp"
+
+        request.data = json.loads(request.body)
+
+        with patch.object(TurnActions.objects, "filter", return_value=[turn_action]):
+            with patch.object(Organization.objects, "get", return_value=self.org):
```

Comment:
Can we not just create the correct objects here? The org definitely exists so that patch doesn't look like we need it